src/JWK.js

```diff
@@ -85,8 +87,18 @@ class JWK {
    */
   static importKey (data, options) {
     return Promise.resolve()
+
+      // Calculate Thumbprint
+      .then(() => this.thumbprint(data))
+
       // Create instance
-      .then(() => new JWK(data, options))
+      .then(hash => {
+        if (hash && !data.kid && !(options && options.kid)) {
```

Comment:
One minor optimization (pointed out by @johnny90) -- maybe it would make sense to only generate the thumbprint if it's not already passed in. So, like:

```js
  // Calculate Thumbpring
  .then(() => {
    if (data.kid || options.kid) {
      return data.kid || options.kid
    } else {
      return this.thumbprint(data)
    }
  })

  // Create instance
  .then(kid => {
    data.kid = kid
    // ... etc
  })
```